src/wp-includes/media.php

```diff
@@ -1547,10 +1550,13 @@ function wp_image_add_srcset_and_sizes( $image, $image_meta, $attachment_id ) {
 				}
 			}
 		}
-	}
 
-	if ( ! $width || ! $height ) {
-		return $image;
+		if ( ! $width || ! $height ) {
+			return $image;
+		}
+
+		// Add width and height if not present.
+		$attr .= ' ' . trim( image_hwstring( $width, $height ) );
```

Comment:
Hmm, I was thinking about that - I'm not opposed, however this would already behave incorrectly with the current behavior. If only one of the two attributes was specified, the existing logic would still set both `$width` and `height` based on the attachment metadata, so all the images in `srcset` and `sizes` would likely not match the aspect ratio of how the image is displayed. I think specifiying only one of the two can essentially be considered not supported by core at this point.